subsys/net/lib/ptp/transport.c

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright (c) 2024 BayLibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ptp_transport, CONFIG_PTP_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/net/socket.h>
+
+#include "transport.h"
+
+#define INTERFACE_NAME_LEN (32)
+
+#define IP_MULTICAST_IP "224.0.1.129"
+#define IP6_MULTICAST_IP "FF0E:0:0:0:0:0:0:181"
```

Comment:
Since these are constant, we could just statically initialize `mcast_addr` accordingly instead of parsing the address string at runtime, IMO it doesn't really impact the readability, for instance:
`in_addr mcast_addr = { { { 224, 0, 1, 129 } } };`